platform/binance/client.go

```diff
@@ -2,83 +2,129 @@ package binance
 
 import (
 	"encoding/json"
+	"fmt"
 	"github.com/trustwallet/blockatlas/pkg/blockatlas"
 	"github.com/trustwallet/blockatlas/pkg/errors"
 	"github.com/trustwallet/blockatlas/pkg/logger"
 	"net/http"
 	"net/url"
 	"strconv"
+	"time"
 )
 
-// TODO Headers + rate limiting
-
 type Client struct {
 	blockatlas.Request
 }
 
-func (c *Client) GetBlockList(count int) (*BlockList, error) {
-	result := new(BlockList)
-	query := url.Values{"rows": {strconv.Itoa(count)}, "page": {"1"}}
-	err := c.Get(result, "blocks", query)
+const (
+	TokensLimit       = "1000"
+	TransactionsLimit = "25"
+)
+
+// Fetch runtime information about the node
+func (c *Client) fetchNodeInfo() (*NodeInfo, error) {
+	result := new(NodeInfo)
+	err := c.Get(result, "v1/node-info", nil)
 	return result, err
 }
 
-func (c *Client) GetBlockByNumber(num int64) (*TxPage, error) {
-	stx := new(TxPage)
+// Get transactions in the block. Multi-send and multi-coin transactions are included as sub-transactions.
+func (c *Client) GetBlockTransactions(num int64) ([]TxV2, error) {
+	stx := new(BlockTransactions)
+	path := fmt.Sprintf("v2/transactions-in-block/%d", num)
```

Comment:
http://dex.binance.org/api/v2/transactions-in-block/80157267?
Not working now

Reply:
Could you please use rpc tx_search instead of use this API? This API is not rpc, still costing a lot of resource 

curl "localhost:27147/tx_search?query=\"tx.height=10000\"&prove=true"
https://github.com/binance-chain/java-sdk/wiki/RPC#query-transactions-by-block
https://docs.binance.org/api-reference/node-rpc.html#5-rpc-endpoint-list